src/transport.c

```diff
@@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
 				break;
 
 		if (matched == node->addrlen) {
-			*address_bits_matched = matched * 8;
-			*me = node;
-			did_match = EXACT_MATCH;
-			break;
+			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
```

Comment:
Does it even build? It appears match_tmp is not declared ;)

And I don't actually get it here. If it's the exact match, why is it not the only one? Unless the node is running as both site and arbitrator?